docs/proposals/devfile.md

```diff
@@ -0,0 +1,64 @@
+
+# Devfile support in odo
+
+To better integrate odo with other developer tooling we need to align on one common definition of a developer workspace and application lifecycle.
+
+[Eclipse Che](https://github.com/eclipse/che) project already created and started using [Devfile](https://github.com/redhat-developer/devfile). Devfile is a YAML file that defines developer workspace in Che environment. It is structured in such a way that can be used also in odo.
+
+Implementing Devfile support in odo will create a great benefit for users, as it will allow them to easily switch tools with no additional configuration. Adding support for new languages will be also much easier with devfiles, as it will be only a matter of using the right devfile template.
+
+## Implementation plan
+
+### Consuming current version of Devfile in odo
+
+1. Create a golang library for parsing and reading data from Devfile
+    * Devfile is versioned file so the design of a library needs to be able to handle and support multiple Devfile versions at the same time
+
+        **Parser high level design**
+        * The parser would read the provided devfile.
+        * From the devfile, parser would look for the API Version of the devfile and validate whether the API version in supported in odo.
+        * The devfile schema is based on <http://json-schema.org/draft-07/schema#>
+        * Hence, the parser would validate the integrity of the provided devfile with the devfile json schema of the respective API version.
+        * Furthermore, parser would try to map devfile sections (projects, components, commands etc) to their respective Golang structs.
```

Comment:
Not sure what you mean by that. We define the odo component as a building block of a bigger application (microservice). In that sense, it will be still the same component

Reply:
I mean that, AFAIK, the word `component` means different things to odo and DevFile. Are we adding any information about that here or do we plan to do that in the docs?

Reply:
There are discussions about renaming this in Devfile. So hopefully it will get renamed in future devfile spec revision.
If not that we will have to address this in docs.
I don't think it needs to be mentioned in this doc.